opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -442,7 +442,7 @@ message HistogramDataPoint {
   // Negative events *can* be recorded, but sum should not be filled out when
   // doing so.  This is specifically to enforce compatibility w/ OpenMetrics,
   // see: https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#histogram
-  double sum = 5;
+  optional double sum = 5;
```

Comment:
Out of curiosity (and not having read any prior context), is there a semantic meaning to optional fields in OTLP? i.e., why would we allow SDKs to not populate this, given it can provide value? Or if it doesn't provide value why do we not just deprecate it for removal?

If it's about some fields being possibly useful if the backend understands it, then it's a concept I haven't seen before in OTel, which generally leans toward sending as much data as possible. If it is about that sort of efficiency though, I think we first need some sort of negotiation protocol defined, where the required fields (and semantic attributes!) are negotiated beforehand before sending - that would then make this sort of optionality make sense.

Reply:
Unfortunately, this one is about Prometheus compatibility.   Prometheus histogram sum only allows monotonic sums, and OpenTelemetry can compute histograms with non-monotonic sums.   This bit of OTLP only exists for prometheus compatibility, specifically knowing that sum-in-histogram can be a sum-in-prometheus.   There are alternative ways to solve this that were discarded.    If you read the description of the proto, this change merely brings this in line with what was decided for the v0.9 version of OTLP.

Reply:
Yeah I figured it's something like that - but since SDKs currently don't know if their data will end up in Prometheus, I think either 1) they always fill it or 2) they are able to negotiate what they are required to fill is needed, isn't it?